### PR TITLE
small fixes

### PR DESCRIPTION
**Pull request by BelehovEgor**
fixed validation of leave time
fixed find stat - only return worktimes, now we return all for user in this month

### CODE REVIEW

File: src/TimeService.Validation/LeaveTime/EditLeaveTimeRequestValidator.cs [hunk 1]
```diff
@@ -1,132 +1,133 @@
...
-            #endregion
-
-            #region IsActive
+      Context = context;
+      RequestedOperation = requestedOperation;
+
+      #region Paths
+
+      AddСorrectPaths(
+          new()
```
Comment (Cheephot) [inline src/TimeService.Validation/LeaveTime/EditLeaveTimeRequestValidator.cs:25]: tab

File: src/TimeService.Validation/LeaveTime/EditLeaveTimeRequestValidator.cs [hunk 2]
```diff
@@ -1,132 +1,133 @@
...
+      AddСorrectOperations(nameof(EditLeaveTimeRequest.LeaveType), new List<OperationType> { OperationType.Replace });
+      AddСorrectOperations(nameof(EditLeaveTimeRequest.IsActive), new List<OperationType> { OperationType.Replace });
+      AddСorrectOperations(nameof(EditLeaveTimeRequest.Comment), new List<OperationType> { OperationType.Replace });
+
+      #endregion
+
+      #region StartTime
+
+      AddFailureForPropertyIf(
+          nameof(EditLeaveTimeRequest.StartTime),
```
Comment (Cheephot) [inline src/TimeService.Validation/LeaveTime/EditLeaveTimeRequestValidator.cs:47]: tab

File: src/TimeService.Validation/LeaveTime/EditLeaveTimeRequestValidator.cs [hunk 3]
```diff
@@ -1,132 +1,133 @@
...
+          {
+            { x => DateTime.TryParse(x.value.ToString(), out _), "Incorrect format of StartTime." },
+          });
+
+      #endregion
+
+      #region EndTime
+
+      AddFailureForPropertyIf(
+          nameof(EditLeaveTimeRequest.EndTime),
```
Comment (Cheephot) [inline src/TimeService.Validation/LeaveTime/EditLeaveTimeRequestValidator.cs:59]: tab

File: src/TimeService.Validation/LeaveTime/EditLeaveTimeRequestValidator.cs [hunk 4]
```diff
@@ -1,132 +1,133 @@
...
+          {
+            { x => DateTime.TryParse(x.value.ToString(), out _), "Incorrect format of EndTime." },
+          });
+
+      #endregion
+
+      #region Minutes
+
+      AddFailureForPropertyIf(
+          nameof(EditLeaveTimeRequest.Minutes),
```
Comment (Cheephot) [inline src/TimeService.Validation/LeaveTime/EditLeaveTimeRequestValidator.cs:71]: tab

File: src/TimeService.Validation/LeaveTime/EditLeaveTimeRequestValidator.cs [hunk 5]
```diff
@@ -1,132 +1,133 @@
...
+          {
+            { x => int.TryParse(x.value.ToString(), out int count) && count > 0, "Incorrect format of Minutes." },
+          });
+
+      #endregion
+
+      #region LeaveType
+
+      AddFailureForPropertyIf(
+          nameof(EditLeaveTimeRequest.LeaveType),
```
Comment (Cheephot) [inline src/TimeService.Validation/LeaveTime/EditLeaveTimeRequestValidator.cs:83]: tab

File: src/TimeService.Validation/LeaveTime/EditLeaveTimeRequestValidator.cs [hunk 6]
```diff
@@ -1,132 +1,133 @@
...
+          {
+            { x => Enum.TryParse(typeof(LeaveType), x.value.ToString(), out _), "Incorrect format of LeaveType." },
+          });
+
+      #endregion
+
+      #region IsActive
+
+      AddFailureForPropertyIf(
+          nameof(EditLeaveTimeRequest.IsActive),
```
Comment (Cheephot) [inline src/TimeService.Validation/LeaveTime/EditLeaveTimeRequestValidator.cs:95]: tab

File: src/TimeService.Validation/LeaveTime/EditLeaveTimeRequestValidator.cs [hunk 7]
```diff
@@ -1,132 +1,133 @@
...
-                new()
+    public EditLeaveTimeRequestValidator(ILeaveTimeRepository repository)
+    {
+      RuleForEach(x => x.Operations)
+        .Custom(HandleInternalPropertyValidation);
+
+      When(x => x.Operations.Any(o => o.path.EndsWith(nameof(EditLeaveTimeRequest.StartTime), StringComparison.OrdinalIgnoreCase))
+        && x.Operations.Any(o => o.path.EndsWith(nameof(EditLeaveTimeRequest.EndTime), StringComparison.OrdinalIgnoreCase)),
+        () =>
+            {
```
Comment (Cheephot) [inline src/TimeService.Validation/LeaveTime/EditLeaveTimeRequestValidator.cs:113]: tab

**Review by Cheephot — APPROVED** [review]

**Review by VLiamin — APPROVED** [review]

File: src/TimeService.Validation/LeaveTime/EditLeaveTimeRequestValidator.cs (hunk 1 above)
Comment (BelehovEgor) [reply src/TimeService.Validation/LeaveTime/EditLeaveTimeRequestValidator.cs]: fixed

File: src/TimeService.Validation/LeaveTime/EditLeaveTimeRequestValidator.cs (hunk 2 above)
Comment (BelehovEgor) [reply src/TimeService.Validation/LeaveTime/EditLeaveTimeRequestValidator.cs]: fixed

File: src/TimeService.Validation/LeaveTime/EditLeaveTimeRequestValidator.cs (hunk 4 above)
Comment (BelehovEgor) [reply src/TimeService.Validation/LeaveTime/EditLeaveTimeRequestValidator.cs]: fixed

File: src/TimeService.Validation/LeaveTime/EditLeaveTimeRequestValidator.cs (hunk 5 above)
Comment (BelehovEgor) [reply src/TimeService.Validation/LeaveTime/EditLeaveTimeRequestValidator.cs]: fixed

File: src/TimeService.Validation/LeaveTime/EditLeaveTimeRequestValidator.cs (hunk 3 above)
Comment (BelehovEgor) [reply src/TimeService.Validation/LeaveTime/EditLeaveTimeRequestValidator.cs]: fixed

File: src/TimeService.Validation/LeaveTime/EditLeaveTimeRequestValidator.cs (hunk 6 above)
Comment (BelehovEgor) [reply src/TimeService.Validation/LeaveTime/EditLeaveTimeRequestValidator.cs]: fixed

File: src/TimeService.Validation/LeaveTime/EditLeaveTimeRequestValidator.cs (hunk 7 above)
Comment (BelehovEgor) [reply src/TimeService.Validation/LeaveTime/EditLeaveTimeRequestValidator.cs:113]: fixed

**Review by MariyaSin — APPROVED** [review]

File: src/TimeService.Business/Commands/LeaveTime/EditLeaveTimeCommand.cs [hunk 8]
```diff
@@ -1,126 +1,131 @@
...
+      _validator = validator;
+      _repository = repository;
+      _mapper = mapper;
+      _accessValidator = accessValidator;
+      _httpContextAccessor = httpContextAccessor;
+    }
+
+    public OperationResultResponse<bool> Execute(Guid leaveTimeId, JsonPatchDocument<EditLeaveTimeRequest> request)
+    {
+      var oldLeaveTime = _repository.Get(leaveTimeId);
```
Comment (Glunky) [inline src/TimeService.Business/Commands/LeaveTime/EditLeaveTimeCommand.cs:85]: var

File: src/TimeService.Validation/LeaveTime/CreateLeaveTimeRequestValidator.cs [hunk 9]
```diff
@@ -62,16 +61,14 @@ private bool CheckUserExistence(List<Guid> userIds)
...
         .NotEqual(new DateTime());
 
-      //TODO rework line 70
-      RuleFor(lt => lt)
-        .Must(lt => lt.StartTime < lt.EndTime).WithMessage("Start time must be before end time.")
-        .Must(lt =>
-        {
-          var leaveTimes = repository.Find(new FindLeaveTimesFilter { UserId = lt.UserId, SkipCount = 0, TakeCount = int.MaxValue }, out _);
+      RuleFor(lt => lt.Minutes)
+        .GreaterThan(0);
```
Comment (Glunky) [inline src/TimeService.Validation/LeaveTime/CreateLeaveTimeRequestValidator.cs:65]: with message?

**Review by Glunky — APPROVED** [review]

File: src/TimeService.Validation/LeaveTime/CreateLeaveTimeRequestValidator.cs (hunk 9 above)
Comment (BelehovEgor) [reply src/TimeService.Validation/LeaveTime/CreateLeaveTimeRequestValidator.cs:65]: the default message is enough here

File: src/TimeService.Business/Commands/LeaveTime/EditLeaveTimeCommand.cs [hunk 10]
```diff
@@ -1,126 +1,131 @@
...
-        public OperationResultResponse<bool> Execute(Guid leaveTimeId, JsonPatchDocument<EditLeaveTimeRequest> request)
+      if (startTimeOperation == null && endTimeOperation == null)
+      {
+        return true;
+      }
+
+      DateTime? start = startTimeOperation == null ? null : DateTime.Parse(startTimeOperation.value.ToString());
+      DateTime? end = endTimeOperation == null ? null : DateTime.Parse(endTimeOperation.value.ToString());
+
+      const string compareTimeError = "Start time must be less than end time.";
```
[the remark this answers is not in the document]
Comment (BelehovEgor) [reply src/TimeService.Business/Commands/LeaveTime/EditLeaveTimeCommand.cs]: fixed

File: src/TimeService.Business/Commands/LeaveTime/EditLeaveTimeCommand.cs (hunk 8 above)
Comment (BelehovEgor) [reply src/TimeService.Business/Commands/LeaveTime/EditLeaveTimeCommand.cs]: fixed

**Pull request merged**